dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/base/memory/RelCo.java

```diff
@@ -183,9 +184,11 @@ protected BooleanFormula encodeApprox(SolverContext ctx) {
                 BooleanFormula execPair = getExecPair(t, ctx);
                 lastCo = bmgr.and(lastCo, bmgr.not(relation));
 
-                IntegerFormula a1 = convertToIntegerFormula(w1.getMemAddressExpr(), ctx);
-                IntegerFormula a2 = convertToIntegerFormula(w2.getMemAddressExpr(), ctx);
-                BooleanFormula sameAddress = imgr.equal(a1, a2);
+                Formula a1 = w1.getMemAddressExpr();
+                Formula a2 = w2.getMemAddressExpr();
+                BooleanFormula sameAddress = ARCH_PRECISION > -1 ? 
+                		bvmgr.equal((BitvectorFormula)a1, (BitvectorFormula)a2) : 
+                		imgr.equal((IntegerFormula)a1, (IntegerFormula)a2);
```

Comment:
Isn't this what `Utils.generalEquals` does?

Reply:
No precisely. This code might potentially be unsafe because the lack of conversions (e.g. `convertToBitvectorFormula`). See the detail comment below.

Reply:
The code snippet has no conversion operators so it will break if there is a type mismatch just like `generalEquals` would.
In fact, it is more likely to break in my eyes.

Reply:
Exactly, since this code does not apply the conversions, it can run into problems (even if our current tests do not expose this).